src/components/TodoItem.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { Todo } from '../types/Todo';
+
+interface TodoItemProps {
+  filteredTodos: Todo[];
+  tempClearAll: Todo[] | null;
+  tempDeleteTodo: Todo | null;
+  handleDelete: (id: number) => void;
+}
+
+export const TodoItem: React.FC<TodoItemProps> = ({
+  filteredTodos,
+  tempClearAll,
+  tempDeleteTodo,
+  handleDelete,
+}) => {
+  return (
+    <>
+      {filteredTodos.map(todo => {
```

Comment:
You shouldn't map in this component. Do it in `TodoList`. TodoItem component should be for one item